tests/test_combine.py

```diff
@@ -0,0 +1,68 @@
+import re
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+import pytest
+import xarray as xr
+
+from openghg.dataobjects import ObsData
+from openghg.combine import combine_and_elevate_inlet, combine_data_objects, combine_multisite
+
+
+def make_obs_dataset(start: str, end: str) -> xr.Dataset:
+    times = pd.date_range(start, end, inclusive="left")  # daily frequency
+    values = np.arange(0, len(times))
+
+    return xr.Dataset({"mf": (["time"], values)}, coords={"time": times})
+
+
+def make_obs_data_object(start: str, end: str, metadata: Optional[dict] = None) -> ObsData:
+    data = make_obs_dataset(start, end)
+
+    if metadata is None:
+        metadata = {}
+
+    obs_dat = ObsData(metadata=metadata, data=data)
+
+    return obs_dat
+
+
+@pytest.fixture()
+def make_obs_data():
+    obs_dat1 = make_obs_data_object("2024-01-01", "2024-02-01", metadata={"test": 1, "inlet": "10m"})
+    obs_dat2 = make_obs_data_object("2024-02-01", "2024-03-01", metadata={"test": 2, "inlet": "11m"})
+    obs_dat3 = make_obs_data_object("2024-03-01", "2024-04-01", metadata={"test": 3, "inlet": "12m"})
+    obs_dat4 = make_obs_data_object("2024-04-01", "2024-05-01", metadata={"test": 4, "inlet": "10m"})
+
+    return obs_dat1, obs_dat2, obs_dat3, obs_dat4
+
+
+def test_combine_data_objects(make_obs_data):
+    data_objects = list(make_obs_data)
+    result = combine_data_objects(data_objects)
+
+    assert result.metadata == {"test": 1, "inlet": "10m"}
+    assert (result.data.time.values == pd.date_range("2024-01-01", "2024-05-01", inclusive="left").values).all()
+
+
+def test_combine_data_objects_by_inlet(make_obs_data):
+    data_objects = list(make_obs_data)
+
+    result = combine_and_elevate_inlet(data_objects)
+
+    expected_values = np.concatenate([x * np.ones(len(ds.data.time.values)) for x, ds in zip([10.0, 11.0, 12.0, 10.0], data_objects)])
+
+    np.testing.assert_equal(result.data.inlet.values, expected_values)
+
+    assert "inlet" not in result.metadata
+
+
+def test_combine_data_objects_by_site():
+    data_objects = [make_obs_data_object("2024-01-01", "2024-02-01", metadata={"site": x}) for x in "abcd"]
+
+    result = combine_multisite(data_objects)
+
+    expected_dataset = xr.concat([do.data.expand_dims({"site": [x]}) for do, x in zip(data_objects, "abcd")], dim="site")
+
+    xr.testing.assert_equal(result.data, expected_dataset)
```

Comment:
I added tests for `combine_multisite` and `combine_and_elevate_inlet` using real data. I had to change how `combine_data_objects` worked to allow a `concat_dim` to be specified because `combine_by_coords` was trying to concatenate on time in `combine_multisite` before concatenating on `site`. I'm not sure if it would always do this, but just concatenating by site works. I think it's a useful option to have because it can share all of the metadata handling.